main.py

```diff
@@ -1,39 +1,104 @@
 import argparse
 import json
 import logging
+import fnmatch
 
 from lm_eval import tasks, evaluator
 
 logging.getLogger("openai").setLevel(logging.WARNING)
 
+class MultiChoice:
+    def __init__(self, choices):
+        self.choices = choices
+
+    # Simple wildcard support (linux filename patterns)
+    def __contains__(self, values):
+        for value in values.split(","):
+            if len(fnmatch.filter(self.choices, value)) == 0:
+                return False
+
+        return True
+
+    def __iter__(self):
+        for choice in self.choices:
+            yield choice
+
+# Get task base classes for filtering
+task_types = list(set([task.__bases__[0].__name__ for task in tasks.TASK_REGISTRY.values()]))
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument('--model', required=True)
     parser.add_argument('--model_args', default="")
-    parser.add_argument('--tasks', default="all_tasks")
+    parser.add_argument('--tasks', default=None, choices=MultiChoice(tasks.ALL_TASKS))
+    parser.add_argument('--task_type', default=None, choices=MultiChoice(task_types))    
+    parser.add_argument('--exclude_tasks', default=None, choices=MultiChoice(tasks.ALL_TASKS))
     parser.add_argument('--provide_description', action="store_true")
     parser.add_argument('--num_fewshot', type=int, default=0)
     parser.add_argument('--batch_size', type=int, default=None)
     parser.add_argument('--device', type=str, default=None)
     parser.add_argument('--output_path', default=None)
     parser.add_argument('--limit', type=int, default=None)
     parser.add_argument('--no_cache', action="store_true")
-    parser.add_argument('--description_dict_path', default=None)
+    parser.add_argument('--decontaminate', action="store_true")
+    parser.add_argument('--ngrams_path', default=None)
+    parser.add_argument('--ngrams_n_size', type=int, default=None)
```

Comment:
same nit: rename stuff to begin with decontamination_

Reply:
Same as before, long arguments, happy to change it if you want?